src/Artsy/Router/buildServerApp.tsx

```diff
@@ -107,14 +130,38 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
             )
           }
 
+          // Build up script tags to inject into head
+          const scripts = []
+
           const { relayData: _relayData, styleTags, bodyHTML } = await trace(
             "buildServerApp.fetch",
             (async () => {
               const sheet = new ServerStyleSheet()
 
-              const html = ReactDOMServer.renderToString(
-                sheet.collectStyles(<ServerApp tags={headTags} />)
-              )
+              // Extract bundle split chunks for SSR
+              let bundleJSX
+
+              // FIXME: Remove once EXPERIMENTAL_APP_SHELL A/B test completes
+              if (getENV("EXPERIMENTAL_APP_SHELL")) {
+                const extractor = new ChunkExtractor({
+                  statsFile,
+                  entrypoints: ["experimental-app-shell"],
+                })
+
+                const bundleScriptTags = extractor.getScriptTags()
+                scripts.concat(bundleScriptTags)
+
+                // Wrap component tree in library contexts to extract usage
+                bundleJSX = extractor.collectChunks(
+                  sheet.collectStyles(<ServerApp tags={headTags} />)
+                )
+
+                // TODO: Wire up if A/B test is successful
+                // extractor.getLinkTags()
```

Comment:
This is in support for "fetch bundle on hover" style behavior, similar to gatsby or instant-page.